packages/api/src/web5.ts

```diff
@@ -239,69 +205,72 @@ export class Web5 {
-                  enc             : '#enc',
-                  sig             : '#sig',
-                }
-              ],
-              verificationMethods: [
-                {
-                  algorithm : 'Ed25519',
-                  id        : 'sig',
-                  purposes  : ['assertionMethod', 'authentication']
-                },
-                {
-                  algorithm : 'secp256k1',
-                  id        : 'enc',
-                  purposes  : ['keyAgreement']
-                }
-              ]
-            }
+      // userAgent.
+      // Connect attempt failed or was rejected so fallback to local user agent.
+      let identity: BearerIdentity;
+
+      // Query the Agent's DWN tenant for identity records.
+      const identities = await userAgent.identity.list();
+
+      // If an existing identity is not found found, create a new one.
+      const existingIdentityCount = identities.length;
+
+      if (existingIdentityCount === 0) {
+        // Use the specified DWN endpoints or get default tech preview hosted nodes.
+        const serviceEndpointNodes = techPreview?.dwnEndpoints ?? await getTechPreviewDwnEndpoints();
+
+        // Generate a new Identity for the end-user.
+        identity = await userAgent.identity.create({
+          didMethod  : 'dht',
+          metadata   : { name: 'Default' },
+          didOptions : {
+            services: [
+              {
+                id              : 'dwn',
+                type            : 'DecentralizedWebNode',
+                serviceEndpoint : serviceEndpointNodes,
+                enc             : '#enc',
+                sig             : '#sig',
+              }
+            ],
+            verificationMethods: [
+              {
+                algorithm : 'Ed25519',
+                id        : 'sig',
+                purposes  : ['assertionMethod', 'authentication']
+              },
+              {
+                algorithm : 'secp256k1',
+                id        : 'enc',
+                purposes  : ['keyAgreement']
+              }
+            ]
+          }
+        });
+
+        if (walletConnectOptions) {
+          WalletConnect.initClient({
+            connectServerUrl : walletConnectOptions.connectServerUrl,
+            walletUri        : walletConnectOptions.walletUri,
+            onUriReady       : walletConnectOptions.onUriReady,
+            permissionRequests:
+            walletConnectOptions.permissionRequests,
+            validatePin: walletConnectOptions.validatePin,
           });
-
-          // The User Agent will manage the Identity, which ensures it will be available on future
-          // sessions.
-          await userAgent.identity.manage({ portableIdentity: await identity.export() });
-
-        } else if (existingIdentityCount === 1) {
-          // An existing identity was found in the User Agent's tenant.
-          identity = identities[0];
-
         } else {
```

Comment:
note for review in case it's not clear the impersonation isn't here yet. i need to work with liran to add it